superset/views/core.py

```diff
@@ -1513,6 +1513,15 @@ def tables(self, db_id, schema, substr, force_refresh='false'):
             table_names = [tn for tn in table_names if substr in tn]
             view_names = [vn for vn in view_names if substr in vn]
 
+        if not schema and database.default_schemas:
+            def get_schema(tbl_or_view_name):
+                return tbl_or_view_name.split('.')[0]
```

Comment:
Do names always have a period in them? The suggestion I gave before handles tables without a schema:

```python
>>> def get_schema(tbl_or_view_name):
...     return tbl_or_view_name.split('.')[0] if '.' in tbl_or_view_name else None
...
>>> get_schema('foo.bar')
'foo'
>>> get_schema('test')
None
```

Your implementation would return `test` in the last call. I think it's fine if the names always contain periods (otherwise, why would you add `default_schemas`?), but I think it's better to be correct just in case.